### PR TITLE
Fixing the maximum number of OMTF final candidates [12_4_X]

### PR DESCRIPTION
**Pull request by kpedro88**
backport of #39624

For Run 3 (2022) MC production with signal models with many muons.

#### PR validation:

Reran a failing MC job (DIGI step) and it now succeeds.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

see above

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

File: L1Trigger/L1TMuonOverlapPhase1/src/Omtf/GhostBuster.cc [hunk 1]
```diff
@@ -32,6 +32,9 @@ AlgoMuons GhostBuster::select(AlgoMuons refHitCands, int charge) {
     }
     if ((*it1)->getQ() > 0 && !isGhost)
       refHitCleanCands.emplace_back(new AlgoMuon(**it1));
+
+    if (refHitCleanCands.size() >= 3)
+      break;
```
Comment (aloeliger) [inline L1Trigger/L1TMuonOverlapPhase1/src/Omtf/GhostBuster.cc:37]: If only for my edification, where is the `3` coming from?

Comment (kpedro88) [reply L1Trigger/L1TMuonOverlapPhase1/src/Omtf/GhostBuster.cc:37]: This is a backport of #39624, so @kbunkow or another muon trigger expert should answer.

**Comment by aloeliger** [comment]
+l1

- Backport of existing fix

**Comment by rappoccio** [comment]
+1

**Pull request merged**